litert/vendors/qualcomm/compiler/Qualcomm_QNN_Compiler.md

```diff
@@ -0,0 +1,132 @@
+
+| LiteRT Quantization Type | Supported | Note |
+| :--- | :--- | :--- |
+| `kLiteRtQuantizationPerTensor` | **Yes** | For `Int4`, this uses `BwScaleOffsetQuantizeParamsWrapper`. |
+| `kLiteRtQuantizationPerChannel` | **Yes** | For `Int4`, this uses `BwAxisScaleOffsetQuantizeParamsWrapper`. Zero points are converted from `int64` to `int32`. |
+| `kLiteRtQuantizationBlockWise` | **No** | Explicitly marked as unsupported. |
+| `kLiteRtQuantizationNone` | **Yes** | |
+
+## Supported Operations
+
+The following `LiteRtOpCode` operations are supported. The "Note" column
+provides the corresponding QNN operation it is legalized to.
+
+| LiteRT Op Code | Note |
+| :--- | :--- |
+| `kLiteRtOpCodeTflAbs` | Legalized to `QNN_OP_ELEMENT_WISE_UNARY` (ABS). |
+| `kLiteRtOpCodeTflAdd` | Legalized to `QNN_OP_ELEMENT_WISE_ADD`. Supports fused activation. |
+| `kLiteRtOpCodeTflArgMax` | Legalized to `QNN_OP_ARGMAX`. |
+| `kLiteRtOpCodeTflArgMin` | Legalized to `QNN_OP_ARGMIN`. |
+| `kLiteRtOpCodeTflAveragePool2d` | Legalized to `QNN_OP_POOL_AVG_2D`. Supports fused activation. |
+| `kLiteRtOpCodeTflBatchMatmul` | Legalized to `QNN_OP_MAT_MUL`. |
+| `kLiteRtOpCodeTflBroadcastTo` | Legalized to `QNN_OP_ELEMENT_WISE_ADD` or `QNN_OP_ELEMENT_WISE_OR` with a static tensor. |
+| `kLiteRtOpCodeTflCast` | Legalized to `QNN_OP_CAST`. |
+| `kLiteRtOpCodeTflCeil` | Legalized to `QNN_OP_ELEMENT_WISE_UNARY` (CEIL). |
+| `kLiteRtOpCodeTflConcatenation` | Legalized to `QNN_OP_CONCAT`. Supports fused activation. |
+| `kLiteRtOpCodeTflConv2d` | Legalized to `QNN_OP_CONV_2D`. Supports fused activation. Weights are transposed. |
+| `kLiteRtOpCodeTflConv3d` | Legalized to `QNN_OP_CONV_3D`. Supports fused activation. |
+| `kLiteRtOpCodeTflCos` | Legalized to `QNN_OP_ELEMENT_WISE_COS`. |
+| `kLiteRtOpCodeTflCumsum` | Legalized to `QNN_OP_CUMULATIVE_SUM`. |
+| `kLiteRtOpCodeTflDepthToSpace` | Legalized to `QNN_OP_DEPTH_TO_SPACE`. |
+| `kLiteRtOpCodeTflDepthwiseConv2d`| Legalized to `QNN_OP_DEPTH_WISE_CONV_2D`. Supports fused activation. Weights are reshaped. |
+| `kLiteRtOpCodeTflDequantize` | Legalized to `QNN_OP_DEQUANTIZE` or `QNN_OP_CAST`. |
+| `kLiteRtOpCodeTflDiv` | Legalized to `QNN_OP_ELEMENT_WISE_DIVIDE`. Supports fused activation. |
+| `kLiteRtOpCodeTflDynamicUpdateSlice`| Legalized to a sequence of ops including `QNN_OP_REDUCE_SUM` and `QNN_OP_ELEMENT_WISE_SELECT`. |
+| `kLiteRtOpCodeTflElu` | Legalized to `QNN_OP_ELEMENT_WISE_NEURON` (ELU). |
+| `kLiteRtOpCodeTflEmbeddingLookup` | Legalized to `QNN_OP_GATHER`. |
+| `kLiteRtOpCodeTflEqual` | Legalized to `QNN_OP_ELEMENT_WISE_BINARY` (EQUAL). |
+| `kLiteRtOpCodeTflExp` | Legalized to `QNN_OP_ELEMENT_WISE_UNARY` (EXP). |
+| `kLiteRtOpCodeTflFloor` | Legalized to `QNN_OP_ELEMENT_WISE_UNARY` (FLOOR). |
+| `kLiteRtOpCodeTflFloorDiv` | Legalized to `QNN_OP_ELEMENT_WISE_BINARY` (FLOOR_DIV). |
+| `kLiteRtOpCodeTflFullyConnected` | Legalized to `QNN_OP_FULLY_CONNECTED`. HTP optimization may convert it to `QNN_OP_CONV_2D`. |
```

Comment:
```suggestion
| `kLiteRtOpCodeTflFullyConnected` | Legalized to `QNN_OP_FULLY_CONNECTED` and `QNN_OP_RESHAPE` if keep_num_dims = true. Supports fused activation.|
```
As for `QNN_OP_CONV_2D`, I think there are some issues with our current HTP optimization flag for `kLiteRtOpCodeTflFullyConnected`. I would prefer not to mention it here.